src/project_types.rs

```diff
@@ -43,3 +56,13 @@ impl PartialEq for Project {
 
 impl Eq for Project {
 }
+
+impl PartialEq for User {
```

Comment:
Get rid of both PartialEq and Eq impls. Add `#![derive(Eq)]` to Project and User.